tests/Backend/CommonPart1/AlterTableTest.php

```diff
@@ -210,6 +210,97 @@ public function testTooManyColumns(): void
         }
     }
 
+    public function testTooManyColumnsCountFailed(): void
+    {
+        $this->skipTestForBackend([
+            self::BACKEND_EXASOL,
+            self::BACKEND_BIGQUERY,
+        ], 'Other backends does have other limits');
+
+        try {
+            $data = [
+                'name' => 'tooManyColumns',
+                'primaryKeysNames' => [],
+                'columns' => [],
+            ];
+            // generate table definition with too many columns - over limit on every backend
+            for ($i = 1; $i <= 5000; $i++) {
+                $data['columns'][] = [
+                    'name' => 'col' . $i,
+                    'definition' => [
+                        'type' => 'BYTEINT',
+                        'nullable' => true,
+                    ],
+                ];
+            }
+            $this->_client->createTableDefinition(
+                $this->getTestBucketId(),
+                $data,
+            );
+            $this->fail('There were 5000 columns man. fail.');
+        } catch (ClientException $e) {
+            $this->assertEquals('storage.tables.validation.tooManyColumns', $e->getStringCode());
+        }
+    }
+
+    public function testTooManyColumnsCountAddAnotherColumnOverLimitFailed(): void
```

Comment:
haha, good point, to me nenapadlo, ze nemam otestovane. Ale poustel jsi to na starych backendech (=snflk, synapse... ? ). imho to ted vyhodi 500, coz nechceme. melo by to podle me vyzkouset vytvorit CollumnsCollection nez se to vubec pusti do backendu. At na to mame jednotne misto, kde to spadne.